src/newgrf.cpp

```diff
@@ -2502,7 +2502,7 @@ static ChangeInfoResult TownHouseChangeInfo(uint hid, int numinfo, int prop, Byt
 				break;
 
 			case 0x17: // Four random colours to use
-				for (uint j = 0; j < 4; j++) housespec->random_colour[j] = buf->ReadByte();
+				for (uint j = 0; j < 4; j++) housespec->random_colour[j] = Colours(buf->ReadByte() & 0xF);
```

Comment:
New magic number without explanation...?

Reply:
Just a mask. I've switched it to `GB(x, 0, 4)` which is still fairly magic, but is used in other places.

Colours is a currently uint8_t but only values 0-15 are valid colours, and 255 is explicitly INVALID_COLOUR, which doesn't make sense to have here. Currently the code doesn't range check, which would end up treating a non-recolour sprite as a recolour sprite.